src/transformers/models/beit/modeling_beit.py

```diff
@@ -211,14 +216,11 @@ def forward(
             embeddings = embeddings * (1 - w) + mask_tokens * w
 
         cls_tokens = self.cls_token.expand(batch_size, -1, -1)
-        if self.position_embeddings is not None:
-            if interpolate_pos_encoding:
-                cls_tokens = cls_tokens + self.interpolate_pos_encoding(embeddings, height, width)
```

Comment:
This is actually a bug, but we probably never reach it because `self.position_embeddings` is None

Reply:
Or it's likely no one never use `interpolate_pos_encoding=True`?

If it is `True`, would it fail the call at this point or it's just compute a different value of `cls_tokens`?

BTW, I see

```
        if config.use_absolute_position_embeddings:
            self.position_embeddings = nn.Parameter(torch.zeros(1, num_patches + 1, config.hidden_size))
        else:
            self.position_embeddings = None
```
So it's possible `self.position_embeddings` is not `None` if `config.use_absolute_position_embeddings` is `True?`

Reply:
Yeah, I checked some popular models on the hub, all have `use_absolute_position_embeddings: false` (didn't do extensive testing tbh). So it's likely the combination of factors: no one uses `use_absolute_position_embeddings=True` (should be a new model) + `interpolate_pos_encoding=True`.

It's a bug introduced with adding `interpolate_pos_encoding` flag, it should be a `embedding = embedding + ..` not `cls_tokens`. But even in that case we will have double interpolatio: here and one in BeitPatchEmbedding, so I just cleaned this up.

Reply:
OK, indeed. Thanks!